include/circt/Support/PrettyPrinter.h

```diff
@@ -156,6 +168,13 @@ struct BeginToken : public TokenBase<BeginToken, Token::Kind::Begin> {
 
 struct EndToken : public TokenBase<EndToken, Token::Kind::End> {};
 
+struct CallbackToken : public TokenBase<CallbackToken, Token::Kind::Callback> {
+  CallbackToken(Token::CallbackInfo::CallbackTy c, unsigned id) {
+    initialize(c, id);
+  }
+  void invoke() const { std::invoke(getInfo().callback, getInfo().id); }
```

Comment:
Should this expose the info members, or is that encapsulation a feature/intentional? Works for me :+1: .